qiskit/algorithms/time_evolvers/variational/var_qte.py

```diff
@@ -0,0 +1,269 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Variational Quantum Time Evolution Interface"""
+from __future__ import annotations
+
+from abc import ABC
+from typing import Type, Callable
+
+import numpy as np
+from scipy.integrate import OdeSolver
+
+from qiskit import QuantumCircuit
+from qiskit.circuit import Parameter
+from qiskit.opflow import PauliSumOp
+from qiskit.primitives import BaseEstimator
+from qiskit.quantum_info.operators.base_operator import BaseOperator
+
+from .solvers.ode.forward_euler_solver import ForwardEulerSolver
+from .solvers.ode.ode_function_factory import OdeFunctionFactory
+from .solvers.ode.var_qte_ode_solver import VarQTEOdeSolver
+from .solvers.var_qte_linear_solver import VarQTELinearSolver
+
+from .variational_principles.variational_principle import VariationalPrinciple
+from .var_qte_result import VarQTEResult
+
+from ..time_evolution_problem import TimeEvolutionProblem
+
+from ...observables_evaluator import estimate_observables
+
+
+class VarQTE(ABC):
+    """Variational Quantum Time Evolution.
+
+    Algorithms that use variational principles to compute a time evolution for a given
+    Hermitian operator (Hamiltonian) and a quantum state prepared by a parameterized quantum
+    circuit.
+
+    References:
+
+        [1] Benjamin, Simon C. et al. (2019).
+        Theory of variational quantum simulation. `<https://doi.org/10.22331/q-2019-10-07-191>`_
+    """
+
+    def __init__(
+        self,
+        ansatz: QuantumCircuit,
+        initial_parameters: dict[Parameter, float] | list[float] | np.ndarray,
+        variational_principle: VariationalPrinciple | None = None,
+        estimator: BaseEstimator | None = None,
```

Comment:
Is None really valid here for these. In _evolve variational_principle is passed directly to a function that seems not to allow None, and estimator is used directly in evaluating operators. To me it looks like None is invalid

Reply:
True, removed None